src/CleanArchitecture.Web/Startup.cs

```diff
@@ -58,14 +51,22 @@ private static IServiceProvider BuildDependencyInjectionProvider(IServiceCollect
             // Populate the container using the service collection
             builder.Populate(services);
 
-            // TODO: Add Registry Classes to eliminate reference to Infrastructure
+            // Reference all CleanArchitecture assemblies for AutoFac.
             Assembly webAssembly = Assembly.GetExecutingAssembly();
             Assembly coreAssembly = Assembly.GetAssembly(typeof(BaseEntity));
-            Assembly infrastructureAssembly = Assembly.GetAssembly(typeof(EfRepository)); // TODO: Move to Infrastucture Registry
+
+            string location = AppDomain.CurrentDomain.BaseDirectory;
+            Assembly infrastructureAssembly = GetInfrastructureAssembly(location);
             builder.RegisterAssemblyTypes(webAssembly, coreAssembly, infrastructureAssembly).AsImplementedInterfaces();
+            builder.RegisterAssemblyModules(infrastructureAssembly);
+            return new AutofacServiceProvider(builder.Build());
+        }
 
-            IContainer applicationContainer = builder.Build();
-            return new AutofacServiceProvider(applicationContainer);
+        private static Assembly GetInfrastructureAssembly(string location)
+        {
+            const string infrastructureDll = "CleanArchitecture.Infrastructure.dll";
```

Comment:
Rather than magic string, you tried this approach? 
```
var infrastructureAssemblyName = AppDomain.CurrentDomain.GetAssemblies()
                .Where(a => a.FullName.Contains("Infrastructure")).Select(x => x.FullName).FirstOrDefault();
```
With that, we don't need to worry about part **CleanArchitecture** when we will change the name of the project.

Reply:
I agree it would be helpful if whatever approach is used here, it survives renaming of the project and its associated DLLs.

Reply:
I actually started by trying to fetch the dynamic infrastructure assembly name via `AppDomain.CurrentDomain.GetAssemblies()`, but `GetAssemblies()` doesn't return assemblies that aren't loaded. And because the Web project does not reference the Infrastructure project anymore, it does not get returned from `GetAssemblies()`.

See the following image:

![image](https://user-images.githubusercontent.com/4809334/53268891-597f3480-36b5-11e9-8626-09069f006d49.png)

After several approaches, referencing the DLL by exact file name was the only way to get the required assembly for AutoFac.

Reply:
Hmm, so then just nameof of the namespace to fail fast.
`
string infrastructureDll = $"{nameof(CleanArchitecture)}.Infrastructure.dll";
`

It will be a pain in the ass if Application Domain will not contain information about Infrastructure, tomorrow I will check something that is related to performance investigation because without that some tools may be broken or inaccurate(I will check that).